src/realm/null.hpp

```diff
@@ -62,11 +62,6 @@ struct null {
     {
         throw(LogicError::type_mismatch);
     }
-    template <class T>
-    operator util::Optional<T>()
-    {
-        return util::none;
-    }
```

Comment:
This removal is why the tests are failing. Deleting the conversion to int64_t instead to resolve the ambiguous conversion seems to work.

Reply:
Thanks for looking into this for me! Done.